test/border_router: Unmodified test with net_tools implementation

#### Description
<!--
Tell us what the bug is about.
Example: Cannot compile for m4a-mb board.
-->
The border Router test wasn't updated with the new implementation of the module net_tools. To resolve this will be changed the main code in the test...
#### Steps to reproduce the issue
<!--
To interactive test
```
make -C /test/border_router flash term
```
then write 's' in the shell to run the test.

Try to describe as precisely as possible how to reproduce the bug,
describing your hardware configuration, OS, or any custom setup also can help.
-->

#### Expected results
<!--
Describe what should happen
-->
```
OK (3 tests)
```
#### Actual results
<!--
Describe what actually happens
-->
```
home/user/Documentsm4a-firmware/tests/border_router/main.c: In function 'test_border_router_add_ipv6':
/home/user/Documentsm4a-firmware/tests/border_router/main.c:33:15: error: implicit declaration of function 'border_router_add_ipv6'; did you mean 'test_border_router_add_ipv6'? [-Werror=implicit-function-declaration]
   33 |     int err = border_router_add_ipv6(_UNICAST,  &address, WIRED_INTERFACE);
      |               ^~~~~~~~~~~~~~~~~~~~~~
      |               test_border_router_add_ipv6
/home/user/Documentsm4a-firmware/tests/border_router/main.c:33:38: error: '_UNICAST' undeclared (first use in this function)
   33 |     int err = border_router_add_ipv6(_UNICAST,  &address, WIRED_INTERFACE);
      |                                      ^~~~~~~~
/home/user/Documentsm4a-firmware/tests/border_router/main.c:33:38: note: each undeclared identifier is reported only once for each function it appears in
/home/user/Documentsm4a-firmware/tests/border_router/main.c: In function 'test_border_router_add_ipv6_node':
/home/user/Documentsm4a-firmware/tests/border_router/main.c:42:38: error: '_UNICAST' undeclared (first use in this function)
   42 |     int err = border_router_add_ipv6(_UNICAST,  &address, WIRELESS_INTERFACE);
```

#### Versions
<!--
Operating system: Mac OSX, Linux, etc
Build environment: GCC, CLang versions, etc
-->

--- a/firmware/network/border_router/Makefile.dep
+++ b/firmware/network/border_router/Makefile.dep
@@ -1,5 +1,3 @@
-
-
 USEMODULE += auto_init_gnrc_netif
 USEMODULE += auto_init_usbus
 USEMODULE += gnrc_icmpv6_echo

--- a/firmware/network/border_router/border_router.c
+++ b/firmware/network/border_router/border_router.c
@@ -39,7 +39,7 @@ int8_t get_wired_iface(void) {
     return -1;
 }
 
-int8_t border_router_setup(ipv6_addr_t addr, uint8_t iface_type) {
+int8_t border_router_setup(ipv6_addr_t addr, uint8_t prefix, uint8_t iface_type) {
     ipv6_addr_t ip;
     int8_t index;
     switch (iface_type) {
@@ -62,13 +62,13 @@ int8_t border_router_setup(ipv6_addr_t addr, uint8_t iface_type) {
         return -1;
     }
     if (ipv6_addr_is_global(&addr)) {
-        if (set_ipv6_global(index, addr) < 0) {
+        if (set_ipv6_global(index, addr, prefix) < 0) {
             return -1;
         }
         return 0;
     }
     if (ipv6_addr_is_multicast(&addr)) {
-        if (set_ipv6_multicast(index, addr) < 0) {
+        if (set_ipv6_multicast(index, addr, prefix) < 0) {
             return -1;
         }
         return 0;

--- a/firmware/network/border_router/include/border_router.h
+++ b/firmware/network/border_router/include/border_router.h
@@ -49,11 +49,11 @@ enum type_iface_t {
  *
  * @param[in] addr       ipv6 address
  * @param[in] iface_type refers to if is used a WIRED or WIRELESS interface.
- *
+ * @param[in] prefix Networks prefix,express the subnet size.
  * @retval 0 Setup Success
  * @retval -1 Setup Failed
  */
-int8_t border_router_setup(ipv6_addr_t addr, uint8_t iface_type);
+int8_t border_router_setup(ipv6_addr_t addr, uint8_t prefix, uint8_t iface_type);
 
 #ifdef __cplusplus
 }

--- a/firmware/network/net_tools/include/net_tools.h
+++ b/firmware/network/net_tools/include/net_tools.h
@@ -59,21 +59,23 @@ int8_t get_ipv6_global(kernel_pid_t iface_pid, ipv6_addr_t *addr);
  * @brief Function to set an ipv6 global address in an iface.
  *
  * @param[in] iface_index  index that will be save the ipv6 address.
- * @param[in] ip Address ipv6 global to set
- * @retval 0 Correctly set up of the @p ip in the interface
- * @retval -1 Couldn't set the @p ip address in the interface
+ * @param[in] ip Address ipv6 global to set.
+ * @param[in] prefix Networks prefix,express the subnet size.
+ * @retval 0 Correctly set up of the @p ip in the interface.
+ * @retval -1 Couldn't set the @p ip address in the interface.
  */
-int8_t set_ipv6_global(kernel_pid_t iface_index, ipv6_addr_t ip);
+int8_t set_ipv6_global(kernel_pid_t iface_index, ipv6_addr_t ip, uint8_t prefix);
 
 /**
  * @brief Function to set an ipv6 global address in an iface.
  *
  * @param[in] iface_index  index that will be save the ipv6 address.
  * @param[in] ip Address ipv6 multicast to set
+ * @param[in] prefix Networks prefix,express the subnet size.
  * @retval  0 Correctly set up of the @p ip in the interface.
  * @retval -1 Couldn't set the @p ip address in the interface.
  */
-int8_t set_ipv6_multicast(kernel_pid_t iface_index, ipv6_addr_t ip);
+int8_t set_ipv6_multicast(kernel_pid_t iface_index, ipv6_addr_t ip, uint8_t prefix);
 
 #ifdef __cplusplus
 }

--- a/firmware/network/net_tools/net_tools.c
+++ b/firmware/network/net_tools/net_tools.c
@@ -56,7 +56,13 @@ int8_t get_ipv6_global(kernel_pid_t iface_pid, ipv6_addr_t *addr) {
     return -1;
 }
 
-int8_t set_ipv6_global(kernel_pid_t iface_index, ipv6_addr_t ip) {
+int8_t set_ipv6_global(kernel_pid_t iface_index, ipv6_addr_t ip, uint8_t prefix) {
+    if (prefix > 128) {
+        printf("Wrong prefix length\n"
+               "File: %s, Function: %s, Line: %d\n",
+               __FILE__, __func__, __LINE__);
+    }
+    uint16_t flags = GNRC_NETIF_IPV6_ADDRS_FLAGS_STATE_VALID | (prefix << 8U);
     netif_t *iface = netif_get_by_id(iface_index);
     if (!ipv6_addr_is_global(&ip)) {
         printf("The ipv6 address isn't a ipv6 global address format accepted\n"
@@ -64,8 +70,7 @@ int8_t set_ipv6_global(kernel_pid_t iface_index, ipv6_addr_t ip) {
                __FILE__, __func__, __LINE__);
         return -1;
     }
-    if (netif_set_opt(iface, NETOPT_IPV6_ADDR, GNRC_NETIF_IPV6_ADDRS_FLAGS_STATE_VALID, &ip,
-                      sizeof(ipv6_addr_t)) < 0) {
+    if (netif_set_opt(iface, NETOPT_IPV6_ADDR, flags, &ip, sizeof(ipv6_addr_t)) < 0) {
         printf("error: unable to add IPv6 address\n"
                "File: %s, Function: %s, Line: %d\n",
                __FILE__, __func__, __LINE__);
@@ -74,15 +79,21 @@ int8_t set_ipv6_global(kernel_pid_t iface_index, ipv6_addr_t ip) {
     return 0;
 }
 
-int8_t set_ipv6_multicast(kernel_pid_t iface_index, ipv6_addr_t ip) {
+int8_t set_ipv6_multicast(kernel_pid_t iface_index, ipv6_addr_t ip, uint8_t prefix) {
+    if (prefix > 128) {
+        printf("Wrong prefix length\n"
+               "File: %s, Function: %s, Line: %d\n",
+               __FILE__, __func__, __LINE__);
+    }
+    uint16_t flags = GNRC_NETIF_IPV6_ADDRS_FLAGS_STATE_VALID | (prefix << 8U);
     netif_t *iface = netif_get_by_id(iface_index);
-    if (ipv6_addr_is_multicast(&ip)) {
+    if (!ipv6_addr_is_multicast(&ip)) {
         printf("The ipv6 address isn't a ipv6 multicast address format accepted\n"
                "File: %s, Function: %s, Line: %d\n",
                __FILE__, __func__, __LINE__);
         return -1;
     }
-    if (netif_set_opt(iface, NETOPT_IPV6_GROUP, 0, &ip, sizeof(ipv6_addr_t)) < 0) {
+    if (netif_set_opt(iface, NETOPT_IPV6_GROUP, flags, &ip, sizeof(ipv6_addr_t)) < 0) {
         printf("error: unable to join IPv6 multicast group\n"
                "File: %s, Function: %s, Line: %d\n",
                __FILE__, __func__, __LINE__);

--- a/tests/border_router/main.c
+++ b/tests/border_router/main.c
@@ -17,36 +17,36 @@
  * @brief       border router test file
  *
  * @author      RocioRojas <Rociorojas391@gmail.com>
+ * @author      eduazocar <eduazocarv@gmail.com>
  */
 
 #include <string.h>
 #include <errno.h>
 #include "embUnit.h"
 #include "border_router.h"
-#include "rpl_protocol.h"
 
-void test_border_router_add_ipv6(void) {
-     ipv6_addr_t address = {
+void test_border_router_setup_wired(void) {
+    ipv6_addr_t address = {
         .u8 = {0},
     };
-    ipv6_addr_from_str(&address, "2001:db8:1::2");
-    int err = border_router_add_ipv6(_UNICAST,  &address, WIRED_INTERFACE);
+    ipv6_addr_from_str(&address, "2001:db8:2::1");
+    int err = border_router_setup(address, 64, WIRED_INTERFACE);
     TEST_ASSERT_EQUAL_INT(0, err);
 }
 
-void test_border_router_add_ipv6_node(void) {
-     ipv6_addr_t address = {
+void test_border_router_setup_wireless(void) {
+    ipv6_addr_t address = {
         .u8 = {0},
     };
     ipv6_addr_from_str(&address, "2001:db8:1::2");
-    int err = border_router_add_ipv6(_UNICAST,  &address, WIRELESS_INTERFACE);
+    int err = border_router_setup(address, 16, WIRELESS_INTERFACE);
     TEST_ASSERT_EQUAL_INT(0, err);
 }
 
 Test *tests_border_router(void) {
     EMB_UNIT_TESTFIXTURES(fixtures){
-        new_TestFixture(test_border_router_add_ipv6),
-        new_TestFixture(test_border_router_add_ipv6_node),
+        new_TestFixture(test_border_router_setup_wired),
+        new_TestFixture(test_border_router_setup_wireless),
     };
 
     EMB_UNIT_TESTCALLER(tests_border_router, NULL, NULL, fixtures);

